blk/components/virt.c

```diff
@@ -123,12 +123,12 @@ static void partitions_init()
     ((blk_storage_info_t *)blk_config)->sector_size = ((blk_storage_info_t *)blk_config_driver)->sector_size;
     ((blk_storage_info_t *)blk_config)->size = clients[0].sectors / (BLK_TRANSFER_SIZE / MSDOS_MBR_SECTOR_SIZE);
     ((blk_storage_info_t *)blk_config)->read_only = false;
-    ((blk_storage_info_t *)blk_config)->ready = true;
+    __atomic_store_n(&((blk_storage_info_t *)blk_config)->ready, true, __ATOMIC_RELEASE);
 #if BLK_NUM_CLIENTS > 1
```

Comment:
Why not do this unconditionally?

Reply:
Just from looking at it, if the client num is less than one, this would result in a bad access to the clients data structure whose length is defined by max blk clients. In the serial and network subsystems, I have moved client number dependent initialisation into a config filed which hides it from the c files. But ultimately I think the goal is to have this system dependent type of initialisation handled by the microkit tool.